### PR TITLE
Tags component

### LINKED ISSUE #13676: UI Component - Tags
This issue is to track the creation of the component:  Tag & Tags 

You can access the links at:
 Tag:   https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=180%3A9685
 Tags: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=181%3A9543

### PR DESCRIPTION
**Pull request by jo-mut**
fixes #13676

### Summary
This PR adds tag & tags component to quo2.preview

### CODE REVIEW

**Review by ibrkhalil — CHANGES_REQUESTED** [review]
Can we please revert the changes in Podfile if not necessary ?

**Review by jo-mut — COMMENTED** [review]
In almost all my PR I have an issue with this podfile. It keeps being mentioned. I think I have to recheck all my branches against develop. Thanks @ibrkhalil

**Comment by ibrkhalil** [comment]
> In almost all my PR I have an issue with this podfile. It keeps being mentioned. I think I have to recheck all my branches against develop. Thanks @ibrkhalil

My pleasure

File: ios/Podfile.lock [hunk 1]
```diff
@@ -210,7 +210,7 @@ PODS:
     - React
   - react-native-blob-util (0.13.18):
     - React-Core
-  - react-native-blur (0.8.0):
+  - react-native-blur (3.6.1):
```
Comment (ibrkhalil) [inline ios/Podfile.lock]:
```suggestion
  - react-native-blur (0.8.0):
```

File: ios/Podfile.lock [hunk 2]
```diff
@@ -635,7 +635,7 @@ SPEC CHECKSUMS:
   FBLazyVector: 3bb422f41b18121b71783a905c10e58606f7dc3e
   FBReactNativeSpec: f2c97f2529dd79c083355182cc158c9f98f4bd6e
   Folly: b73c3869541e86821df3c387eb0af5f65addfab4
-  glog: 502275bc0747df92346edbab1f40dacde2ec86d4
+  glog: 36ce0530c6d2c3a5a4326885ef4069564887a1db
```
Comment (ibrkhalil) [inline ios/Podfile.lock]:
```suggestion
  glog: 502275bc0747df92346edbab1f40dacde2ec86d4
```

File: ios/Podfile.lock [hunk 3]
```diff
@@ -653,7 +653,7 @@ SPEC CHECKSUMS:
   React-jsinspector: 58aef7155bc9a9683f5b60b35eccea8722a4f53a
   react-native-background-timer: 1f7d560647b40e6a60b01c452ba29c54bf581fc4
   react-native-blob-util: 600972b1782380a5a7d5db61a3817ea32349dae9
-  react-native-blur: cad4d93b364f91e7b7931b3fa935455487e5c33c
+  react-native-blur: 4568dd93d1d82748c180df8beb2d929c97b13b47
```
Comment (ibrkhalil) [inline ios/Podfile.lock]:
```suggestion
  react-native-blur: cad4d93b364f91e7b7931b3fa935455487e5c33c
```

File: src/quo2/components/filter_tag.cljs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+          :background-color   background-color
+          :border-width       1
+          :border-radius      size}
+         (when disabled
+           {:opacity 0.3})
+         (if label
+           {:padding-horizontal (case size 32 12 24 8)}
+           {:width              size})))
+
+(defn base-tag [_]
```
Comment (flexsurfer) [inline src/quo2/components/filter_tag.cljs]: probably we should move base-tag in separate ns ?

File: src/quo2/components/filter_tag.cljs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+       [rn/image {:source emoji
+                  :style  (merge (case size
+                                   32 {:height 20
+                                       :width  20}
+                                   24 {:height 12
+                                       :width  12})
+                                 (when label
+                                   {:margin-right 4}))}])
+     (when label
+       [rn/view
```
Comment (flexsurfer) [inline src/quo2/components/filter_tag.cljs]: why extra view ?

File: src/quo2/components/filter_tag.cljs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  [_ _]
+  (fn [{:keys [id on-press disabled size emoji icon active accessibility-label label]
+        :or   {size 32}}]
+    (let [state (cond disabled :disabled active :active :else :default)
+          {:keys [icon-color border-color background-color text-color]}
+          (get-in themes [(theme/get-theme) state])]
+      [rn/touchable-without-feedback (merge {:disabled            disabled
+                                             :accessibility-label accessibility-label}
+                                            (when on-press
+                                              {:on-press (fn []
```
Comment (flexsurfer) [inline src/quo2/components/filter_tag.cljs]: `#(on-press id)`

File: src/quo2/components/filter_tag.cljs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                                                           (on-press id))}))
+       [rn/view
```
Comment (flexsurfer) [inline src/quo2/components/filter_tag.cljs]: why extra view ?

File: src/quo2/components/filter_tags.cljs [hunk 8]
```diff
@@ -0,0 +1,26 @@
+(ns quo2.components.filter-tags
+  (:require [reagent.core :as reagent]
+            [quo.react-native :as rn]
+            [quo2.components.filter-tag :as tag]))
+
+(defn tags [{:keys [default-active on-change]}]
+  (let [active-tab-id (reagent/atom default-active)]
+    (fn [{:keys [data size type with-label disabled] :or {size 32}}]
```
Comment (flexsurfer) [inline src/quo2/components/filter_tags.cljs]: why not just `label` ?

Comment (flexsurfer) [reply src/quo2/components/filter_tags.cljs]: why `with-` ?

File: src/quo2/components/permission_tag.cljs [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+   :height           size
+   :border-radius    size
+   :justify-content  :center
+   :align-items      :center})
+
+(defn tag-resources [tokens size]
+  (let [tkns (take (if (> (count tokens) 3) 2 3) tokens)]
+    (for [{:keys [token-icon]} tkns]
+      ^{:key token-icon}
+      [react/view {:flex-direction :row}
```
Comment (flexsurfer) [inline src/quo2/components/permission_tag.cljs]: why extra view with :row ?

File: src/quo2/components/permission_tag.cljs [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+                     :style         {:align-items :center}}
+          (str "+" extra-counts)]]]])))
+
+(defn tag-or-clause []
+  [react/view {:align-items   :center}
+   [text/text {:weight        :medium
+               :style         {:margin-right 12
+                               :margin-left  4
+                               :font-size     11}}
+    "or"]])
```
Comment (flexsurfer) [inline src/quo2/components/permission_tag.cljs]: translation

File: src/quo2/components/filter_tags.cljs (hunk 8 above)
Comment (jo-mut) [reply src/quo2/components/filter_tags.cljs]: i will change this

File: src/quo2/components/filter_tag.cljs (hunk 7 above)
Comment (jo-mut) [reply src/quo2/components/filter_tag.cljs]: selecting a tag to show the pressed state doesn't seem to work if the base-tag is not wrapped in a view. Thats why

Comment (flexsurfer) [reply src/quo2/components/filter_tag.cljs]: but we don't know why it doesn't work ? :)

Comment (jo-mut) [reply src/quo2/components/filter_tag.cljs]: yeah. I dont know why. I will investigate further

Comment (jo-mut) [reply src/quo2/components/filter_tag.cljs]: a component that composes a View isn't enough to make TouchableWithoutFeedback work. The content inside a TouchableWithoutFeedback have to be wrapped with View. That seems to be the issue and thats why it works when wrapped with a View

**Pull request closed without merging**